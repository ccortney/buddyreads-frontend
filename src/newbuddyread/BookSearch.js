import { Box, TextField, Button } from "@mui/material"

/** Book search component.
 *
 * Rendered by NewBuddyRead component
 *
 * This component doesn't *do* the searching, but it renders the search
 * form and calls the `handleSearchSubmit` function prop that runs in 
 * NewBuddyRead component to do the searching.
 *
 *  Dashboard => {NewBuddyRead}  -> {BookSearch}
 */

const BookSearch = ({handleSearchChange, handleSearchSubmit, searchField}) => {
    return (
        <Box align='center' sx={{justifyContent:'center'}}>
            <form onSubmit={handleSearchSubmit}>
                <Box sx={{ display: 'flex', justifyContent:  'center' }}>
                    <TextField
                        label="Search for a Book"
                        id="searchField"
                        name="searchField"
                        value={searchField}
                        onChange = {handleSearchChange}
                        sx={{ m: 1, width: '25ch' }}
                    />
                    <Box sx={{display: 'flex', alignItems: 'center',  m: 1}}>
                        <Button variant="contained" type="submit">Search</Button>
                    </Box>
                </Box>
            </form>
        </Box>
    )
}

export default BookSearch;