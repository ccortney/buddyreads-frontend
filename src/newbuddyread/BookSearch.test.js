import { render } from "@testing-library/react";
import BookSearch from "./BookSearch";

it("matches snapshot", function () {
  const { asFragment } = render(<BookSearch />);
  expect(asFragment()).toMatchSnapshot();
});
