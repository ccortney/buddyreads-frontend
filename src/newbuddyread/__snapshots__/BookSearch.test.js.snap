// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    align="center"
    class="MuiBox-root css-1tyndxa"
  >
    <form>
      <div
        class="MuiBox-root css-1l4w6pd"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-ekfslj-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="searchField"
            id="searchField-label"
          >
            Search for a Book
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="searchField"
              name="searchField"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1ftyaf0"
              >
                <span>
                  Search for a Book
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiBox-root css-kgh0nc"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Search
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
